orca-igor/src/main/java/com/netflix/spinnaker/orca/igor/BuildService.java

```diff
@@ -18,15 +18,22 @@
 import com.netflix.spinnaker.kork.artifacts.model.Artifact;
 import java.util.List;
 import java.util.Map;
-import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 import org.springframework.web.util.UriUtils;
 import retrofit.client.Response;
 
 @Component
-@RequiredArgsConstructor
 public class BuildService {
   private final IgorService igorService;
+  private final boolean jobNameAsQueryParameter;
+
+  public BuildService(
+      IgorService igorService,
+      @Value("${jobName.queryParameter:false}") boolean jobNameAsQueryParameter) {
```

Comment:
A separate configuration class is usually cleaner than sprinkling `@Value` annotations around.  See e.g. [this example](https://github.com/spinnaker/orca/blob/master/orca-web/src/main/java/com/netflix/spinnaker/config/EnhancedMonitoringConfigurationProperties.java), enabled [here](https://github.com/spinnaker/orca/blob/master/orca-web/src/main/java/com/netflix/spinnaker/config/EnhancedMonitoringConfiguration.java#L39).

Reply:
Okay, I will create a separate properties class and extract the bean definition to a configuration class

Reply:
Thanks.